### PR TITLE
Fluxor dependency optimization

### PR DESCRIPTION
**Pull request by buvinghausen**
The big change is when targeting TFM .net5.0 you get the 5.x dependencies and .net6.0 you get the 6.x dependencies.  It also cleans up the dependency tree by supporting .NETs dependency inheritance

### CODE REVIEW

File: Source/Fluxor/Fluxor.csproj [hunk 1]
```diff
@@ -6,11 +6,6 @@
     <PackageTags>Redux Flux DotNet CSharp</PackageTags>
   </PropertyGroup>
 
-  <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
```
Comment (mrpmorris) [inline Source/Fluxor/Fluxor.csproj:10]: Why are netstandard 2.0 and 2.1 no longer supported?

File: Source/Fluxor.Blazor.Web/Fluxor.Blazor.Web.csproj [hunk 2]
```diff
@@ -7,14 +7,13 @@
...
-    <LangVersion>9</LangVersion>
     <RazorLangVersion>3.0</RazorLangVersion>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components" Version="3.1.8" />
-    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="3.1.8" />
+    <PackageReference Condition="'$(TargetFramework)' == 'net6.0'" Include="Microsoft.AspNetCore.Components.Web" Version="6.0.0" />
+    <PackageReference Condition="'$(TargetFramework)' == 'net5.0'" Include="Microsoft.AspNetCore.Components.Web" Version="5.0.0" />
+    <PackageReference Condition="'$(TargetFramework)' != 'net6.0' AND '$(TargetFramework)' != 'net5.0'" Include="Microsoft.AspNetCore.Components.Web" Version="3.1.0" />
```
Comment (mrpmorris) [inline Source/Fluxor.Blazor.Web/Fluxor.Blazor.Web.csproj:16]: Don't we also need AspNetCore.Components for 3.1?

File: Source/Directory.Build.props [hunk 3]
```diff
@@ -25,4 +27,9 @@
 		<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
 		<NoWarn>NU5118</NoWarn>
 	</PropertyGroup>
+
+	<ItemGroup>
+		<SupportedPlatform Include="browser" />
```
Comment (mrpmorris) [inline Source/Directory.Build.props:32]: Should this be here? Fluxor.csproj is UI agnostic.

**Review by mrpmorris — COMMENTED** [review]
Thanks for your PR!

I've added a few questions regarding your proposed changes.

Also, would you mind creating a ticket for this, please?

File: Source/Fluxor/Fluxor.csproj (hunk 1 above)
Comment (buvinghausen) [reply Source/Fluxor/Fluxor.csproj:10]: They are I just hoisted the dependencies up into your Directory.Build.props so you only have one spot to make changes going forward.

File: Source/Directory.Build.props (hunk 3 above)
Comment (buvinghausen) [reply Source/Directory.Build.props:32]: All it is is a safeguard to ensure that no committer accidentally uses code marked with the following [attribute](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.versioning.unsupportedosplatformattribute)

```csharp
[System.Runtime.Versioning.UnsupportedOSPlatform("browser")]
```

By putting this element into the Directory.Build.props you just ensure you will get a compiler error if anyone were to use a class/function decorated as such in one of their edits in all three of your libraries.

If you create a blazorwasm hosted project you can see that the Microsoft template puts the element in the csproj on the shared csproj even though it is not exclusively a WASM class library.

```
dotnet new blazorwasm -ho -o BlazorDemo
```

File: Source/Fluxor/Fluxor.csproj (hunk 1 above)
Comment (mrpmorris) [reply Source/Fluxor/Fluxor.csproj:10]: Re: netstandard 2.0 and 2.1, do the Blazor libraries still compile with those there?

Comment (buvinghausen) [reply Source/Fluxor/Fluxor.csproj:10]: Indeed.

image

image

**Pull request merged**